format-xls/src/main/java/io/cdap/plugin/format/xls/input/XlsInputFormatProvider.java

```diff
@@ -0,0 +1,192 @@
+      context.getFailureCollector().addFailure("Sheet number must be a number.", null)
+              .withConfigProperty(XlsInputFormatConfig.NAME_SHEET_VALUE);
+    }
+    FailureCollector collector = context.getFailureCollector();
+    if (!conf.containsMacro(PathTrackingConfig.NAME_SCHEMA) && schema == null && context.getInputSchema() == null) {
+      collector.addFailure("XLS format cannot be used without specifying a schema.", "Schema must be specified.")
+              .withConfigProperty(PathTrackingConfig.NAME_SCHEMA);
+    }
+  }
+
+  @Override
+  protected void addFormatProperties(Map<String, String> properties) {
+    properties.put(XlsInputFormat.SHEET_NO, conf.getSheet());
+    properties.put(XlsInputFormat.SHEET_VALUE, conf.getSheetValue());
+    properties.put(XlsInputFormat.NAME_SKIP_HEADER, String.valueOf(conf.getSkipHeader()));
+    properties.put(XlsInputFormat.TERMINATE_IF_EMPTY_ROW, String.valueOf(conf.getTerminateIfEmptyRow()));
+    properties.put(FileInputFormat.SPLIT_MINSIZE, Long.toString(Long.MAX_VALUE));
+  }
+
+  @Override
+  @Nullable
+  public Schema detectSchema(FormatContext context, InputFiles inputFiles) throws IOException {
+    String blankHeader = "BLANK";
+    FailureCollector failureCollector = context.getFailureCollector();
+    FormulaEvaluator formulaEvaluator;
+    for (InputFile inputFile : inputFiles) {
+      DataFormatter formatter = new DataFormatter();
+      try (Workbook workbook = WorkbookFactory.create(inputFile.open())) {
+        formulaEvaluator = workbook.getCreationHelper().createFormulaEvaluator();
+        formulaEvaluator.setIgnoreMissingWorkbooks(true);
+        Sheet workSheet;
+        // Check if user wants to access with name or number
+        if (conf.getSheet() != null && conf.getSheet().equals(XlsInputFormatConfig.SHEET_NUMBER)) {
+          workSheet = workbook.getSheetAt(Integer.parseInt(Objects.requireNonNull(conf.getSheetValue())));
+        } else {
+          workSheet = workbook.getSheet(conf.getSheetValue());
+        }
+
+        // If provided sheet does not exist, throw an exception
+        if (workSheet == null) {
+          failureCollector.addFailure("Sheet " + conf.getSheetValue() + " does not exist in the workbook.",
+                  "Specify a valid sheet.");
+          return null;
+        }
+
+        int sampleSizeInt = getSampleSizeInt();
+        // Row numbers are 0 based in POI
+        int rowStart = Math.min(0, workSheet.getFirstRowNum());
+        int rowEnd = Math.min(sampleSizeInt, workSheet.getLastRowNum());
+
+        int lastCellNumMax = 0;
+        List<String> columnNames = new ArrayList<>();
+        XlsInputFormatSchemaDetector schemaDetector = new XlsInputFormatSchemaDetector();
+        for (int rowIndex = rowStart; rowIndex <= rowEnd; rowIndex++) {
+          Row row = workSheet.getRow(rowIndex);
+          if (row == null) {
+            continue;
```

Comment:
come up with some logic to ignore the usage of `continue` and break down the header & data processing into separate methods for better readability.